libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -693,23 +693,35 @@ void AP_GPS::update(void)
                 // handle switch between real GPSs
                 for (uint8_t i=0; i<GPS_MAX_RECEIVERS; i++) {
                     if (i == primary_instance) {
-                        continue;
+                        continue; 
                     }
                     if (state[i].status > state[primary_instance].status) {
                         // we have a higher status lock, or primary is set to the blended GPS, change GPS
-                        primary_instance = i;
-                        _last_instance_swap_ms = now;
-                        continue;
+                        
+                        if (state[primary_instance].status < GPS_OK_FIX_3D) {
+                            primary_instance = i;
+                            _last_instance_swap_ms = now;
+                            continue;
+                        }
+                        // switch only if the currently used GPS has a 2D FIX or lower
+
+                        if (state[i].status > GPS_OK_FIX_3D) {
+                            primary_instance = i;
+                            _last_instance_swap_ms = now;
+                            continue;
+                        }
+                        // AND only if the other GPS has a DGPS FIX (the best possible fix)
                     }
 
+            /*
```

Comment:
Why was this logic removed?

Reply:
The ideology behind the new logic is that there is no need to switch unless it is absolutely necessary to. Because the second GPS is installed as a back-up unit, to be used unless the primary GPS fails or gets really inaccurate.
So in order to ascertain that ideology, there seems to be no reason to use 'no. of satellites' as a factor for switching, rather, using just the 'FIX type' is more than enough to achieve our goal while maintaining flight safety and significantly lower no. of switching between the two GPS.